### PR TITLE
Implement search center indicator for Unified Map

### LINKED ISSUE #7783: Have marker for search center
From support mail (translated):
It is a nice feature, that I can now directly open the map from the address search result, however sometimes it can be helpful to get a marker on the map, which shows the coordinates of the search result / search center to help with orientation on the map.

### PR DESCRIPTION
**Pull request by moving-bits**
## Description
Implements search center indicator for UnifiedMap.
In conjunction with #14449 this fixes #7783.

(BTW: First time for me working with the new UM layers and UM viewmodel - quite nice to work with!)

### CODE REVIEW

File: main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapViewModel.java [hunk 1]
```diff
@@ -53,6 +56,11 @@ public void setCurrentPositionAndHeading(final Location location, final float he
         positionAndHeading.setValue(new Pair<>(location, heading));
     }
 
+    /** null if coords indicator should be hidden */
+    public void setCoordsIndicator(@Nullable final Geopoint geopoint) {
```
Comment (fm-sys) [inline main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapViewModel.java:60]: This setter method might not be necessary here. You could just call `viewmodel.coordsIndicator.setValue(geopoint);` from inside UnifiedMapActivity... At least that's the way how I did it at most other places. Explicit setter methods inside the viewmodel should IMHO just be used if there is more processing involved but not for simple pass through.

File: main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapViewModel.java [hunk 2]
```diff
@@ -20,6 +20,8 @@
 import androidx.lifecycle.MutableLiveData;
 import androidx.lifecycle.ViewModel;
 
+import javax.annotation.Nullable;
```
Comment (fm-sys) [inline main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapViewModel.java:23]: IIRC we use `androidx.annotation.Nullable` normally. Not sure what's the difference thought... (but it'll get obsolete anyway if we remove the setter again)

**Review by fm-sys — COMMENTED** [review]
Thank you very much for trying it out, LGTM!

I've added one suggestion though...

File: main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapViewModel.java (hunk 1 above)
Comment (moving-bits) [reply main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapViewModel.java:60]: ah, that's even easier, thanks for pointing this out. I'm still in the process of getting used to the new ViewModel for UnifiedMap...

File: main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapViewModel.java (hunk 2 above)
Comment (moving-bits) [reply main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapViewModel.java:23]: Yeah, don't know why I've select this one from the suggestions (or whether it got added automatically). Anyway, is removed meanwhile.

**Comment by moving-bits** [comment]
> I've added one suggestion though...

Thanks for cross-checking. Updated my PR accordingly.

File: main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapViewModel.java (hunk 1 above)
Comment (fm-sys) [reply main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapViewModel.java]: > ah, that's even easier, thanks for pointing this out. I'm still in the process of getting used to the new ViewModel for UnifiedMap...

Really appreciated! Don't hesitate to ask if any questions should arise :)

**Pull request merged**